source/Cosmos.Core/CPU.cs

```diff
@@ -11,15 +11,33 @@ namespace Cosmos.Core
     /// <summary>
     /// CPU class. Non hardware class, only used by core and hardware drivers for ports etc.
     /// </summary>
-    public class CPU
+    public static class CPU
     {
+        public static ulong totalRam;
         // Amount of RAM in MB's.
         // needs to be static, as Heap needs it before we can instantiate objects
         /// <summary>
-        /// Get amount of RAM in MB's. Plugged.
-        /// </summary>
-        [PlugMethod(PlugRequired = true)]
-        public static uint GetAmountOfRAM() => throw null;
+        /// Get amount of RAM in MB's.
+        /// </summary>
+        public static uint GetAmountOfRAM()
+        {
+            if (MemoryMapExists())
+            {
+                var memoryMap = GetMemoryMap();
+                for (int i = 0; i < memoryMap.Length; i++)
+                {
+                    if (memoryMap[i].Type == 1 && memoryMap[i].Length != 0)

```

Comment:
this `memoryMap[i].Length != 0` is not need (and just needs more clock cycles)
as if it is 0 then it would just be `totalRam = totalRam + 0` as X + 0 = X

Reply:
Seems unneccesary ofc but there are reasons why we may need it.